### ISSUE TITLE
ShallowLike parents are root blocks

### ISSUE THREAD

**Issue by alexsporn** [iotaledger/iota-core#553]
```
node-3-validator-1  | 2023-11-24T11:02:38Z	WARN	Protocol	app@v0.0.0-20231113110812-4ca2b6cc9a42/component.go:152	Booker BlockInvalid, blockID: BlockID(0x48b90daa59818a58ac7b64f3121c0b3887a18e9473e62ec835fe73f7508259ed3f000000:63), error: %!w(<nil>): shallow like parent BlockID(0x45c50dfc3a1e453c4b1c044fbb46cc79bf72c4fed600ec07b7aadba20d34887138000000:56) does not contain a conflicting transaction: failed to inherit conflicts for block BlockID(0x48b90daa59818a58ac7b64f3121c0b3887a18e9473e62ec835fe73f7508259ed3f000000:63): failed to book block
```

**Comment by karimodm** [comment]
If something is a rootblock it means that the slot containing the commitment has been committed, therefore the transaction carried by the block should be either accepted, rejected or pending. In this latter case the conflict should have been dropped: we should not allow a conflict to be carried forward indefinitely, so we should "forget" about upon committing.
TL;DR: do not mark the block as invalid if a shallow like points to a rooblock. But let's still enforce that such block must contain a TX, otherwise mark the block as Invalid.

**Comment by karimodm** [comment]
Take a look at https://github.com/iotaledger/iota-core/pull/399
